This endpoint can be used to get the breakeven 2pt percentage needed for a 2pt shot to be better than a 3pt shot for a given percentage

In [1]:
import requests

### NBA - down 3 with 20 seconds left and a 30% 3pt shot

In [2]:
league = "nba"
margin = -3 # from the perspective of the team with the ball
seconds_remaining = 20 # This is time remaining after the shot is made or rebounded
three_point_pct = 30

url = f"https://api.pbpstats.com/get-endgame-breakeven-2-pct/{league}/{margin}/{seconds_remaining}/{three_point_pct}"

response = requests.get(url)
response_json = response.json()
response_json

{'breakeven2_pct': 55}

A 55% 2pt shot is equivalent to a 30% 3pt shot in terms of the impact on your win probability